src/parser/DASHTree.cpp

```diff
@@ -124,7 +124,27 @@ bool adaptive::CDashTree::Open(std::string_view url,
 
   MergeAdpSets();
 
-  m_currentPeriod = m_periods[0].get();
+  auto& kodiProps = CSrvBroker::GetKodiProps();
+  //! @todo: can have sense to move period selection on PostInit or another place
+  //! just before the session initialize period, that will be a common code for all manifest types,
+  //! also because the code related to live delay calculation on AdaptiveStream::start_stream
+  //! can potentially fall on the previous period and so its needed to set in advance the right period
+  //! so live delay and period selection could be merged
+  //! to do this its needed to test also the behaviour of HLS streams with discontinuities
+  uint64_t now = stream_start_ - available_time_;
+  if (m_isLive && !kodiProps.IsPlayTimeshift())
+  {
+    for (auto& period : m_periods)
+    {
+      if (period->GetStart() != NO_VALUE && now >= period->GetStart())
+        m_currentPeriod = period.get();
```

Comment:
worth putting  a break here?

Reply:
no because you need to find the last period that fall in to "now" not the first one